src/realm/parser/parser_utils.hpp

```diff
@@ -73,13 +80,50 @@ std::string key_path_to_string(const KeyPath& keypath);
 StringData get_printable_table_name(StringData name);
 StringData get_printable_table_name(const Table& table);
 
-template<typename T>
-T stot(std::string const& s) {
+// Converts ascii c-locale uppercase characters to lower case,
+// leaves other char values unchanged.
+inline char toLowerAscii(char c)
+{
+    if (isascii(c) && isupper(c))
+        return _tolower(c);
+    return c;
+}
+
+// Looks for +-infinity, NaN
+// There is spotty support for these edge cases on some platforms
+// so we implement manual checks here
+template <typename T>
+bool try_parse_specials(std::string str, T& ret)
+{
+    std::transform(str.begin(), str.end(), str.begin(), toLowerAscii);
+    if (std::numeric_limits<T>::has_quiet_NaN && (str == "nan" || str == "+nan" || str == "-nan")) {
+        ret = std::numeric_limits<T>::quiet_NaN();
```

Comment:
Don't we need to negate this if `str[0] == '-'` to preserve the sign of the nan?

Reply:
I'll change it for consistency, but I can't think of any case where it would matter? `NaN == NaN` is false right, so `NaN == -NaN` would also be false in any case. It's really strange though; even if a float or double column contains `NaN` values, querying for them specifically will return no results because they don't compare equal. That's our current implementation at least. I could see an argument being made for fixing queries to actually find `NaN` values; if we did that, we'd need to think carefully about whether a distinction should be made between `-NaN` vs `NaN`.